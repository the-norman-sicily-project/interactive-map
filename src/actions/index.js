import _ from 'lodash';
import set from 'lodash/fp/set';

export const INIT_MAP = 'INIT_MAP';
export const LOAD_SITES_BEGIN = 'LOAD_SITES_BEGIN';
export const LOAD_SITES_SUCCESS = 'LOAD_SITES_SUCCESS';
export const LOAD_SITES_FAILURE = 'LOAD_SITES_FAILURE';

export const initMap = () => {
  return {
    type: INIT_MAP,
  };
};

export const loadSitesBegin = () => {
  return {
    type: LOAD_SITES_BEGIN,
  };
};

export const loadSitesSuccess = result => {
  const sites = _.get(result, 'features', []).map((feature, index) =>
    set('id', `site_${index + 1}`, feature)
  );

  return {
    type: LOAD_SITES_SUCCESS,
    sites,
  };
};

export const loadSitesFailure = error => {
  return {
    type: LOAD_SITES_FAILURE,
    error,
  };
};
