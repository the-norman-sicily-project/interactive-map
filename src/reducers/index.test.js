import { mapReducer } from './index';
import { INIT_MAP, LOAD_SITES_BEGIN, LOAD_SITES_SUCCESS, LOAD_SITES_FAILURE } from '../actions';

describe('mapReducer', () => {
  describe('INIT_MAP', () => {
    test('returns the correct state', () => {
      const action = { type: INIT_MAP };
      const state = { loading: false, sites: [1, 2, 3] };
      expect(mapReducer(state, action)).toMatchSnapshot();
    });
  });

  describe('LOAD_SITES_BEGIN', () => {
    test('returns the correct state', () => {
      const action = { type: LOAD_SITES_BEGIN };
      const state = { loading: false, sites: [1, 2, 3] };
      expect(mapReducer(state, action)).toMatchSnapshot();
    });
  });

  describe('LOAD_SITES_SUCCESS', () => {
    test('returns the correct state', () => {
      const action = { type: LOAD_SITES_SUCCESS, sites: [4, 5, 6] };
      const state = { loading: true, sites: [1, 2, 3] };
      expect(mapReducer(state, action)).toMatchSnapshot();
    });
  });

  describe('LOAD_SITES_FAILURE', () => {
    test('returns the correct state', () => {
      const action = {
        type: LOAD_SITES_FAILURE,
        error: new Error('Some error occurred'),
      };
      const state = { loading: true, sites: [1, 2, 3] };
      expect(mapReducer(state, action)).toMatchSnapshot();
    });
  });
});
