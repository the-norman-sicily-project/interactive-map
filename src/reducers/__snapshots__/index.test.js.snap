// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer INIT_MAP returns the correct state 1`] = `
Object {
  "loading": false,
  "sites": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`rootReducer LOAD_SITES_BEGIN returns the correct state 1`] = `
Object {
  "loading": true,
  "sites": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`rootReducer LOAD_SITES_FAILURE returns the correct state 1`] = `
Object {
  "error": [Error: Some error occurred],
  "loading": false,
  "sites": Array [],
}
`;

exports[`rootReducer LOAD_SITES_SUCCESS returns the correct state 1`] = `
Object {
  "loading": false,
  "sites": Array [
    4,
    5,
    6,
  ],
}
`;
